bundles/CoreBundle/Command/Document/MigrateElementsCommand.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Command\Document;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use Pimcore\Console\AbstractCommand;
+use Pimcore\Db;
+use Pimcore\Document\Tag\NamingStrategy\NamingStrategyInterface;
+use Pimcore\Document\Tag\NamingStrategy\NestedNamingStrategy;
+use Pimcore\Model\Document\Targeting\TargetingDocumentInterface;
+use Pimcore\Model\Element\Recyclebin\Item;
+use Pimcore\Model\Version;
+use Pimcore\Tool\Serialize;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+class MigrateElementsCommand extends AbstractCommand
+{
+    /**
+     * @var bool
+     */
+    private $runCommand = true;
+
+    protected function configure()
+    {
+        $this
+            ->setName('pimcore:documents:migrate-elements')
+            ->setDescription('Migrates document elements to editables. See issue https://github.com/pimcore/pimcore/issues/7384 first');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (!$this->runCommand) {
+            return 0;
+        }
+
+        $db = Db::get();
+        $versionsRaw = $db->fetchAll("SELECT v.id AS vId, d.id as dId, d.key as `dKey` FROM versions v, documents d WHERE ctype = 'document' AND v.cid = d.`id` AND (d.`type`  = 'snippet' OR d.`type` = 'page')");
+
+        foreach ($versionsRaw as $versionRaw) {
+            $this->processVersionRow($versionRaw);
+        }
+
+        $recycleBinItemsRaw = $db->fetchAll("SELECT * FROM recyclebin WHERE `type` = 'document' ");
+        foreach ($recycleBinItemsRaw as $recycleBinItemRaw) {
+            $this->processRecycleBinItemRow($recycleBinItemRaw);
```

Comment:
do we really need to migrate recyclebin items for Pimcore X? since it is required to flush recyclebin items prior update. please see  https://github.com/pimcore/pimcore/pull/8562/files#diff-17fa19e18a07e264f92f2ac76fcbdde62f76da12b34984abf0e8b40c7baffd72R9

Reply:
I would say let's drop it. wdyt